BSP/Simulator/Src/BSP_Timer.c

```diff
@@ -0,0 +1,78 @@
+#include "BSP_Timer.h"
+
+#define FILE_NAME DATA_PATH(TIMER_CSV)
+
+static void (*Timer1Handler)(void);
+static void (*Timer2Handler)(void);
+int TempValue = 0xFFFFFF; //variable to store reload value written to CSV previously because everyhtime soemthing new is written previous values are erased
+
+
+/** 
+* @brief   Updates the time by reading the CSV file and calling relevant functions at the right time 
+* @param   none
+* @return  none
+*/  
+void BSP_Timer_Update()
+{
+    FILE *file = fopen(FILE_NAME, "r");
+    int fno = fileno(file);
+    flock(fno, LOCK_EX);
+    int timeCurrent[2];
+    int reload;
+    int index=0;
+    
+    for(int i=0;i<2;i++)
+    {
+        fscanf(file,"%d , %d",timeCurrent+index,&reload);
+        index++;
+    }
+    flock(fno, LOCK_UN);
+    fclose(file);
+ 
+
+     if(timeCurrent[0]==0)
+      {
+          Timer1Handler();
+      }
+      if(timeCurrent[1]==0)
+      {
+          Timer2Handler(); 
+      }
+  }
+
+/** 
+* @brief   Intializes the Timer
+* @param   Timer Reload value 
+* @param   Function Pointer
+* @param   Specefic Timer (Timer 1 = 0 and Timer 2 = 1)
+* @return  none
+ */  
+
+void BSP_Timer_Init(int TimerReload,void *func,Time_t time)
+{
+  if(time==Time1)
+  {
+    FILE *file = fopen(FILE_NAME,"w");
+    int fno = fileno(file);
+    flock(fno, LOCK_EX);
+    fprintf(file,"%d,%d",TimerReload,TimerReload);
+    fprintf(file,"\n%d,%d",TempValue,TempValue);

```

Comment:
This will overwrite the value that may have been already initialized here. You should instead first read the values that are currently written to the file, write new values to the timer that was specified, and write the value that was read to the other timer. See BSP_CAN for that approach.

Reply:
Since this will happen, I created a variable call TempValue that stores the value of the previous Reload value and prints out the value again on the CSV so nothing is lost. It worked  when I ran it.

Reply:
I see what you've done now after looking closer. The only problem with that is that this approach assumes that the 2 timers are always going to be initialized immediately one after the other. If one is initialized, more code is run, then the other is initialized, the CURRENT register of the first initialized timer will be changed back to its original value, which will mess with the interrupt frequency during initialization. I would prefer if you read the values and rewrote the line that is not being changed.